pkg/genericapiserver/enablement_config.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package genericapiserver
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+// APIResourceConfigSource is the interface to determine which versions and resources are enabled
+type APIResourceConfigSource interface {
+	AnyVersionOfResourceEnabled(resource unversioned.GroupResource) bool
+	ResourceEnabled(resource unversioned.GroupVersionResource) bool
+	AllResourcesForVersionEnabled(version unversioned.GroupVersion) bool
+	AnyResourcesForVersionEnabled(version unversioned.GroupVersion) bool
+}
+
+// Specifies the overrides for various API group versions.
+// This can be used to enable/disable entire group versions or specific resources.
+type GroupVersionResourceConfig struct {
+	// Whether to enable or disable this entire group version.  This dominates any enablement check.
```

Comment:
Maybe that can find its way into the comment, if true :)

On Thu, Mar 24, 2016 at 11:05 AM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/genericapiserver/enablement_config.go
> <https://github.com/kubernetes/kubernetes/pull/23067#discussion_r57361088>
> :
>
> > +	"k8s.io/kubernetes/pkg/api/unversioned"
> > +	"k8s.io/kubernetes/pkg/util/sets"
> > +)
> > +
> > +// APIResourceConfigSource is the interface to determine which versions and resources are enabled
> > +type APIResourceConfigSource interface {
> > +	AnyVersionOfResourceEnabled(resource unversioned.GroupResource) bool
> > +	ResourceEnabled(resource unversioned.GroupVersionResource) bool
> > +	AllResourcesForVersionEnabled(version unversioned.GroupVersion) bool
> > +	AnyResourcesForVersionEnabled(version unversioned.GroupVersion) bool
> > +}
> > +
> > +// Specifies the overrides for various API group versions.
> > +// This can be used to enable/disable entire group versions or specific resources.
> > +type GroupVersionResourceConfig struct {
> > +	// Whether to enable or disable this entire group version.  This dominates any enablement check.
>
> I *think* Enable=true means the group version is enabled, and
> EnabledResources/DisabledResources are considered. Enable=false means the
> group version is disabled, and EnabledResources/DisabledResources are not
> considered.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23067/files/1dba81ae05eee970f37e2e104b27f16685836b17#r57361088>
>